scripts/run_cycif_glxy_wf.sh

```diff
@@ -27,10 +33,117 @@ if [ ${#sample_files[@]} -eq 0 ]; then
     exit 1
 fi
 
+# Define parameters that change by sample
+# Start by identifying lab
+if [[ $sample_dir =~ CHIN ]]; then
+    lab=CHIN
+elif [[ $sample_dir =~ HMS ]]; then
+    lab=HMS 
+elif [[ $sample_dir =~ COUSSENS ]]; then
+    echo Error - have no implemented support for COUSSENS lab data
+    exit 1
+    lab=COUSSENS
+else
+    echo Error - no lab identified for sample directory \n $sample_dir
+    exit 1
+fi
+# Get the metadata files based on lab
+if [ -z "$assay" ]; then
+    assay=HMS_Immune_v1
+    echo Using default assay: $assay
+else
+    echo Assay provided: $assay
+fi
+
+# Find the sample files
+# Define sort_files
+# Description: Given the lab, sort the files
+# Invocation: sort_files ${file_arr[@]}
+function sort_files {
+    local paths=( `ls $@` )
+    local base_tmp
+    # Define sorting/filtering strategy based on lab
+    if [[ $lab =~ CHIN ]]
+    then
+        local qc_removed=( )
+        for file in ${paths[@]}; do
+            base_tmp=`basename $file`
+            if [ ${base_tmp[@]:2:1} = Q ]; then
+                continue
+            elif [ ${base_tmp[@]:3:1} = Q ]; then
+                continue
+            else
+                echo Using file $file
+                qc_removed[${#qc_removed[@]}]=$file
+            fi
+        done
+        cat << EOF
+$( echo ${qc_removed[@]} | tr " " "\n" | sed 's/\(.*\)\//\1\t/' | sort -V -k 2 | sed 's/\t/\//' )
+EOF
+    else 
+        echo Error: Sorting strategy not defined 0>2
+        exit 1
+    fi 
+}
+# Define find_sorted_samples
+# Description: Determine strategy and find files in sorted order for sample path
+# Invocation: find_sorted_function path/to/sample_dir [glob_pattern]
+function find_sorted_files {
+    local path=$1
+    local glob=$2
+
+    # Determine strategy for finding samples
+    # If Chin lab sample
+    if [[ $path =~ CHIN ]]
+    then
+        local final_glob=CZI*/R*${glob}*.${filetype}
+        sort_files CHIN ${path}/${final_glob}
+
+    elif [[ $path =~ HMS ]]
+    then 
+        sample_files=$(find $path -name '*rcpnl')
+        cat << EOF                                                               
+$( echo ${sample_files[@]} | tr " " "\n" | sed 's/\(.*\)\//\1\t/' | sort -V -k 2 | uniq | sed 's/\t/\//' )
+EOF
+
+    elif [[ $path =~ COUSSENS ]]
+    then 
+        >&2 echo No strategy defined for COUSSENS lab
+        exit 1
+    elif [[ $path =~ KDL ]]
+    then
+        >&2 echo No strategy defined for KDL
+        exit 1
+    else
+        >&2 echo Lab not recognized - $lab
+        exit 1
+    fi 
+}
+echo Finding files from $sample_dir
+sample_files=( $(find_sorted_files ${sample_dir} ${extra_pattern}) )
```

Comment:
One idea here: We could instead have a flag for split-scene. If a user does not want to split by scene, all files will be run under a single history (and merged together). If a user wants to split by scene, then we would check file names to see if the `Scene` is included in the names. If it is, loop through the last steps to create a separate config per sample/scene and create separate histories for each sample/scene. Thoughts?

Reply:
Hmm, it seems like it might not be worth the effort, since I _think_ that we've discovered that ashlar works properly with multi-scene input files, meaning we shouldn't encounter new split scene samples, correct? I think there are two existing samples that this affects, so I lean towards thinking that the hacky glob pattern will suffice?